frontend/device/generic/device.lua

```diff
@@ -499,13 +499,35 @@ function Device:retrieveNetworkInfo()
             result = result .. "SSID: " .. util.trim(ssid) .. "\n"
             std_out:close()
         end
-        if os.execute("ip r | grep -q default") == 0 then
+        local gw = nil
+        -- first try ip route
+        std_out = io.popen('2>/dev/null ip r')
+        if std_out then
+            local routes = std_out:read("*all")
+            for line in util.gsplit(routes, "\n", false) do
+                gw = gw or line:match("^default via (%d+.%d+.%d+.%d+)")
+            end
+            std_out:close()
```

Comment:
Don't the native Lua per-line iterator do the job here?

Reply:
I think I figured it out, please have a look at the updated version.

Reply:
Nope, I meant something much simpler like:

```lua
local routes = io.popen("ip r 2>/dev/null")
for line in routes:lines() do
-- do stuff with line
end
```

Reply:
Can you please point me to the documentation? I'm new to Lua and I wasn't able to find anything like that.

Reply:
That's be the io module docs in the Lua 5.1 reference manual ;).